### PR TITLE
[Utility] Deprecate getExistingVisitLabels()

### LINKED ISSUE #5056: Multiple get Visit Label functions in Utility
There are several function in Utility class that returns a list of Visit labels, we need to clean up the code and use one function.

From redmine https://redmine.cbrain.mcgill.ca/issues/4291

### PR DESCRIPTION
**Pull request by johnsaigle**
## Brief summary of changes

There were two functions for getting visit labels, `getExistingVisitLabels()` and `getVisitList()`. 

This PR replaces the contents of `getVisitList()` with that of `getExistingVisitLabels()`. That's because the former has a better name (more concise + consistent with other functions in the class), but the latter has more robust code, including the option to filter by project.

#### Testing instructions (if applicable)

This function is covered by unit tests so that should suffice.

#### Links to related tickets (GitHub, Redmine, ...)

* resolves #5056
* Similar cleanup done in #5780

### CODE REVIEW

File: php/libraries/Utility.class.inc [hunk 1]
```diff
@@ -747,41 +764,14 @@ class Utility
      * @return array of study visits in the format array('VL' => 'VL')
      *         where VL is the visit label
      *
-     * @note       THIS FUNCTION USES STRING CONCATENATION TO GENERATE AN
-     *             SQL STATEMENT.
-     *             IT IS VULNERABLE TO AN SQL INJECTION ATTACK.
```
Comment (johnsaigle) [inline php/libraries/Utility.class.inc]: I removed this comment because I do not think it is vulnerable:

* Only fixed strings are use to extend the query. 
* pselect is used which should still prepare the only part of the query that can be supplied by a user
* The `int` type hint will throw an error on anything that isn't numeric or null and I don't think it's possible to create any kind of SQL injection attack with integers only.

File: modules/survey_accounts/php/addsurvey.class.inc [hunk 2]
```diff
@@ -61,6 +61,26 @@ class AddSurvey extends \NDB_Form
     {
         $db = \Database::singleton();
 
+        // Check that all required fields are present.
```
Comment (PapillonMcGill) [inline modules/survey_accounts/php/addsurvey.class.inc]: @johnsaigle Are you sure change to this file and  `modules/survey_accounts/test/survey_accountsTest.php` goes into this PR? Look like it should be in another PR.

File: php/libraries/Utility.class.inc (hunk 1 above)
Comment (driusan) [reply php/libraries/Utility.class.inc]: The comment was likely written before it was switched to pselect..

(Even if it wasn't, changing the function to just throw an exception unconditionally also makes it non-vulnerable..)

Comment (johnsaigle) [reply php/libraries/Utility.class.inc]: Fair enough, I figured it was probably a historical thing. I figured somebody would ask about it so I wanted to lay out why this change is safe.

**Comment by johnsaigle** [comment]
@PapillonMcGill @driusan I removed the changes to survey accounts.

**Comment by driusan** [comment]
This is both failing Travis and needs rebase.

File: php/libraries/Utility.class.inc [hunk 3]
```diff
@@ -742,42 +758,14 @@ class Utility
...
-            $ExtraProject_Criteria ORDER BY Visit_label";
-        $visitlabels    = $db->pselect($query, $qparams);
-        $visits         = array();
-        $numVisitLabels = count($visitlabels);
-        for ($i = 0; $i < $numVisitLabels; $i++) {
-            $visits[$visitlabels[$i]['Visit_label']]
-                = ($visitlabels[$i]['Visit_label']);
-        }
-        return $visits;
+        throw new \DeprecatedException(
```
Comment (HenriRabalais) [inline php/libraries/Utility.class.inc:765]: If there isn't any part of the code using this function, do we even need to keep it here to display this error message?

Comment (johnsaigle) [reply php/libraries/Utility.class.inc:765]: I think we do for the case where a project is using it in their custom code.
